api/v1alpha3/webhook_suite_test.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/cluster-api/test/helpers"
+	ctrl "sigs.k8s.io/controller-runtime"
+	// +kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+var (
+	testEnv *helpers.TestEnvironment
+	ctx     = ctrl.SetupSignalHandler()
+)
+
+func TestMain(m *testing.M) {
+	// Bootstrapping test environment
+	utilruntime.Must(AddToScheme(scheme.Scheme))
+	testEnv = helpers.NewTestEnvironment()
+	go func() {
+		if err := testEnv.StartManager(ctx); err != nil {
+			panic(fmt.Sprintf("Failed to start the envtest manager: %v", err))
+		}
+	}()
+	<-testEnv.Manager.Elected()
+	testEnv.WaitForWebhooks()
```

Comment:
I'm not sure this test is enough to avoid this issue appear in the future consider the issue was happening because of missing the old scheme in the runtime (which is initialized in the main.go). I mean when someone removes the old scheme (in the main.go) in the future these tests cannot cover that.

Reply:
@fabriziopandini have you any suggestions for the integration test?

Reply:
I think that create a v1alpha3 object, read it as a v1alpha4 already provides a goo signal about web hooks working.
We should get similar tests for the other API groups

Reply:
+1, I don't think we need to even do any type of validation on the returned resources (other than maybe the returned resource is actually a v1alpha4 resource), since the round trip tests should already be validating that the conversions are functioning.

Reply:
got it, yes I'll add tests for the other API groups